tensorflow/c/while_loop_test.cc

```diff
@@ -33,13 +33,13 @@ class CApiWhileLoopTest : public ::testing::Test {
     TF_DeleteStatus(s_);
   }
 
-  void Init(int ninputs) {
+  void Init(int ninputs, const tensorflow::TensorShape shape = {}) {
```

Comment:
Can you actually make this a PartialTensorShape? That's the only way we can get coverage of unknown-rank inputs (TensorShape = {} corresponds to a scalar, but PartialTensorShape() is unknown rank).

Once you do that, manually specify a scalar shape for BasicLoop or one of the other tests so we get coverage of known shapes working too.